arwen/host/execution.go

```diff
@@ -596,6 +601,42 @@ func (host *vmHost) callSCMethodIndirect() error {
 	return err
 }
 
+func (host *vmHost) revertESDTTransfer(input *vmcommon.ContractCallInput) {
+	if input.Function != core.BuiltInFunctionESDTTransfer {
+		return
+	}
+	if len(input.Arguments) < 2 {
+		return
+	}
+
+	revertInput := &vmcommon.ContractCallInput{
+		VMInput: vmcommon.VMInput{
+			CallerAddr:     input.RecipientAddr,
+			Arguments:      input.Arguments[:2],
+			CallValue:      big.NewInt(0),
+			CallType:       vmcommon.DirectCall,
+			GasPrice:       input.GasPrice,
+			GasProvided:    input.GasProvided,
+			GasLocked:      0,
+			OriginalTxHash: input.OriginalTxHash,
+			CurrentTxHash:  input.CurrentTxHash,
+			ESDTValue:      big.NewInt(0),
+			ESDTTokenName:  nil,
```

Comment:
If this is meant to revert an ESDT transfer, shouldn't this have the original call's ESDTValue and ESDTToken name?

Reply:
it does not matter. built in function does not get it from esdtvalue / esdttokenname but from function and arguments.